### PR TITLE
feat(bat): use the DFT-truncated v2.4 embedding backbone (drop-in)

### PR DESCRIPTION
**Pull request by tphakala**
## What

Repoints the shared BattyBirdNET bat embedding backbone in the model catalog to the DFT-truncated variant `birdnet-v2.4-embeddings-fp32-dfttrunc.onnx` (58,763,257 bytes). It keeps both outputs (`output` `[6522]` at index 0, the `[1024]` embedding at index 1) in the same order, so it is a drop-in that needs no inference-code change: only the catalog checksum, size, and remote path change. It is bit-exact with the current backbone and about 2x faster on CPU (roughly 8 MB smaller).

`RemotePath` points at the new file while `LocalName` stays `birdnet-v24-embeddings.onnx`.

## Why keep the local filename

The shared embeddings file is used by every BattyBirdNET regional model. Keeping `LocalName` stable means:

- Existing installs keep their on-disk `shared/birdnet-v24-embeddings.onnx`. The startup scan is presence-only and never re-verifies the embeddings checksum, so nothing is flagged as corrupt and the bat pipeline keeps working (the two backbones are bit-exact).
- They upgrade to the faster file transparently the next time a bat model is reinstalled or another bat region is installed (the checksum mismatch triggers a normal re-download of the same local filename).
- No orphaned file is left on disk. Renaming the local file would have stranded the old ~67 MB backbone.

New installs fetch the faster file immediately.

## Validation

Parity checked on aarch64 (Raspberry Pi 5) against the current backbone, on both backends the bat embedding extractor uses:

- ONNX Runtime CPU: old vs new `max|delta| = 0.0` on embedding and logits, top-1 identical.
- OpenVINO CPU (fp32): the truncated graph compiles and runs, and old vs new `max|delta| = 0.0`, top-1 identical.

The published file was verified against the catalog constants: HuggingFace reports size 58,763,257 and a content hash matching `embeddingsSHA256`.

Full `golangci-lint` is clean and `go test -race ./internal/classifier/...` passes. The catalog test now pins the new remote path, size, and checksum as literals so an accidental revert fails the test.

## Summary by CodeRabbit

* **Bug Fixes**
  * Updated the bundled BatNET v2.4 embeddings download to the correct DFT-truncated variant.
  * Preserved the existing local filename for compatibility with current installations.
  * Corrected the associated file size and checksum metadata to ensure reliable downloads and validation.

### CODE REVIEW

File: internal/classifier/model_catalog.go [hunk 1]
```diff
@@ -185,9 +185,16 @@ var EmbeddedCatalog = []CatalogEntry{
 }
 
 // Shared BirdNET v2.4 embeddings model, used by all BattyBirdNET classifiers.
+// This is the DFT-truncated variant (remote birdnet-v2.4-embeddings-fp32-dfttrunc.onnx):
+// bit-exact with the original 2-output backbone (embedding max|delta| = 0.0, identical
+// output order, so no inference-code change) but about 2x faster on CPU and roughly 8 MB
+// smaller. The local filename is deliberately kept as birdnet-v24-embeddings.onnx so
+// existing installs keep their on-disk shared/birdnet-v24-embeddings.onnx (the startup
+// scan is presence-only and never re-verifies it) and upgrade transparently the next
+// time a bat model is reinstalled or another bat region is installed.
```
[the remark this answers is not in the document]
Comment (tphakala) [reply internal/classifier/model_catalog.go]: Reworded in 699af4c5. The comment no longer implies the shared embeddings file is presence-checked at startup; it now states the scan only stats each bat model's own regional file and never inspects or re-verifies the shared embeddings file.
